website/server/models/group.js

```diff
@@ -750,15 +750,18 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     $inc: {memberCount: -1},
   };
 
-  let challenges = await Challenge.find({
-    _id: {$in: user.challenges},
-    group: group._id,
-  });
+  // only remove user from challenges if it's set to remove-all
+  if (keep === 'remove-all') {
```

Comment:
Hmm... See, in prod now, you always leave the challenge and the `keep` query param determines if you keep the tasks or have them removed. I _think_ the change you're doing is technically a breaking change.

I think it makes _more_ sense to provide a body parameter for keeping the challenges (in which case the query param is ignored). Then we just update the web app's calls to provide that parameter. 

3rd parties using this route will still remove all the challenges, but it won't be a breaking change.